### DISCUSSION TITLE
go2tv on MacOS ARM ?

### DISCUSSION THREAD

**Discussion by malcomhamelin** [alexballas/go2tv#55] [General]
Hello,

I didn't find anything about it, but I see there's a arm version for Linux, so do you think a MacOS ARM version is possible ? I've not been able to open it with the amd64 one (and it's not about the "the developer cannot be verified" error, it's just "This app cannot be opened")

**Comment by alexballas** [comment]
Hello @malcomhamelin!

This is not really an issue with the build but with the fact that the mac build I share is not signed. This is something I also documented here
https://github.com/alexballas/go2tv/blob/main/README.md

> MacOS potential issue
> 
> If you get the "cannot be opened because the developer cannot be verified" error, you can apply the following workaround.
> 
> - Control-click the app icon, then choose Open from the shortcut menu.
> - Click Open.

After the "Catalina" version, Apple made it even harder to open applications that were not signed with their notarization process. Distributing Mac software requires the usage of a Mac device. On top of that, notarizing the application requires you to be enrolled in the Apple Developer Program for $99/year. Unfortunately I'm not an owner of a Mac device.

These are practically the reasons why I'm not distributing any signed Mac builds (yet)

**Reply by malcomhamelin** [reply]
Hello!

Thanks for the explanation. But yeah I've tried the workaround, and it doesn't seem to work on mine, that's why I was wondering about an ARM version. Because I don't get the usual "the developer cannot be verified" popup, I get a much more definitive version :
![Capture d’écran 2022-09-09 à 15 13 48](https://user-images.githubusercontent.com/39401347/189358348-97578b92-ea66-4f7f-9406-3cba88bb7c6a.png)

Which means "Impossible to open the app".
Usually, when I use an app that is not signed, It still tells me "the developer cannot be verified" and I can use the control click to open an app. Do you think the workaround should still work ?

**Comment by alexballas** [comment]
I created an arm64 build. Would it be possible to test it and let me know if it works for you?
[go2tv_v1.12.0_macOS_arm64.zip](https://github.com/alexballas/go2tv/files/9538630/go2tv_v1.12.0_macOS_arm64.zip)

**Reply by aisamu** [reply]
I've used the ARM build successfully - thanks a lot for this!
```
/.../bin/go2tv: Mach-O 64-bit executable arm64
```

And, if you want one more data point, it streamed to an LG 55UK6360PSF!

**Comment by malcomhamelin** [comment]
Yes of course!
Unfortunately, it can't be opened too and I get this message : 
<img width="307" alt="Capture d’écran 2022-09-10 à 08 50 42" src="https://user-images.githubusercontent.com/39401347/189472771-d706e816-d990-40e6-ba76-d710962b15c7.png">
"app is broken and can't be opened. You should put this element in the bin."

**Comment by alexballas** [comment]
Thank you. I'll try to rent one M1 instance in a cloud provider to run some tests and will let you know.

**Comment by alexballas** [comment]
~@malcomhamelin
~I managed to get my hands on a M1 Mac Mini VM and I manually built the following build. It should work fine.
I still need to investigate why cross compilation fails.
[go2tv.zip](https://github.com/alexballas/go2tv/files/9542592/go2tv.zip)~

**Comment by alexballas** [comment]
I got to the bottom of this. Both arm builds I provided should work fine. The error "is damaged and can’t be opened" seem to be related with this https://osxdaily.com/2019/02/13/fix-app-damaged-cant-be-opened-trash-error-mac/

After I followed step 4, the application opened fine. 
`xattr -rc go2tv.app`
Before I run this command the file had the attribute `com.apple.quarantine`
